cmd/kops/get_cluster.go

```diff
@@ -107,36 +109,57 @@ func (c *GetClustersCmd) Run(args []string) error {
 			return strings.Join(zoneNames, ",")
 		})
 		return t.Render(clusters, os.Stdout, "NAME", "CLOUD", "ZONES")
-	} else if output == OutputYaml {
+
+	case OutputYaml:
+
 		if c.FullSpec {
-			var fullSpecs []*api.Cluster
-			for _, cluster := range clusters {
-				configBase, err := registry.ConfigBase(cluster)
-				if err != nil {
-					return fmt.Errorf("error reading full cluster spec for %q: %v", cluster.Name, err)
-				}
-				fullSpec := &api.Cluster{}
-				err = registry.ReadConfigDeprecated(configBase.Join(registry.PathClusterCompleted), fullSpec)
-				if err != nil {
-					return fmt.Errorf("error reading full cluster spec for %q: %v", cluster.Name, err)
-				}
-				fullSpecs = append(fullSpecs, fullSpec)
+			var err error
+			clusters, err = fullClusterSpecs(clusters)
+			if err != nil {
+				return err
 			}
-			clusters = fullSpecs
 		}
 
 		for _, cluster := range clusters {
-			y, err := api.ToVersionedYaml(cluster)
-			if err != nil {
-				return fmt.Errorf("error marshaling yaml for %q: %v", cluster.Name, err)
+			if err := marshalToStdout(cluster, marshalYaml); err != nil {
+				return err
 			}
-			_, err = os.Stdout.Write(y)
+		}
+		return nil
+	case OutputJSON:
```

Comment:
Probably cleaner to do (pseudocode):

```
case OutputJSON, OutputYaml:
  f := marshalYaml
  if mode == OutputJSON {
    f = marshalJSON
  }

...
```

to avoid the code repetition (though I see you've got rid of most of it with fullClusterSpecs.